cms/core/views.py

```diff
@@ -68,14 +76,97 @@ def liveness(request: "HttpRequest") -> HttpResponse:
                 result = cursor.fetchone()
 
             if result != (1,):
-                return HttpResponseServerError(content=f"'{connection.alias}' database returned unexpected value.")
-        except DatabaseError:
-            return HttpResponseServerError(content=f"'{connection.alias}' database connection errored unexpectedly.")
+                return HttpResponseServerError(f"Database {connection.alias} returned unexpected result")
+        except Exception:  # pylint: disable=broad-exception-caught
+            logger.exception("Database %s reported an error", connection.alias)
+            return HttpResponseServerError(f"Database {connection.alias} reported an error")
 
     if isinstance(caches["default"], RedisCache):
         try:
             get_redis_connection().ping()
-        except redis.exceptions.ConnectionError:
-            return HttpResponseServerError(content="Failed to connect to cache.")
+        except Exception:  # pylint: disable=broad-exception-caught
+            logger.exception("Unable to ping Redis")
+            return HttpResponseServerError("Unable to ping Redis")
 
     return HttpResponse(status=200)
+
+
+@never_cache
+@require_GET
+def health(request: "HttpRequest") -> HttpResponse:
+    now = timezone.now().replace(microsecond=0)
+    data = {
+        "version": {
+            "build_time": settings.BUILD_TIME.isoformat() if settings.BUILD_TIME else None,
+            "git_commit": settings.GIT_COMMIT,
+            "language": "python",
+            "language_version": platform.python_version(),
+            "version": settings.TAG,
+        },
+        "uptime": round((now - settings.START_TIME).total_seconds() * 1000),
+        "start_time": settings.START_TIME.isoformat(),
+    }
+    checks = []
+
+    for connection in connections.all():
+        failed = False
+        message = "Database is ok"
+        try:
+            with connection.cursor() as cursor:
+                cursor.execute(DB_HEALTHCHECK_QUERY)
+                result = cursor.fetchone()
+
+            if result != (1,):
+                failed = True
+                message = "Backend returned unexpected result"
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            failed = True
+            if isinstance(e, DatabaseError):
+                message = "Backend failed"
+            else:
+                logger.exception("Unexpected error connection to backend %s", connection.alias)
+                message = "Unexpected error"
+
+        checks.append(
+            {
+                "name": f"{connection.alias} database",
+                "status": "CRITICAL" if failed else "OK",
+                "status_code": 500 if failed else 200,
+                "message": message,
+                "last_checked": now.isoformat(),
+                "last_success": now.isoformat() if not failed else None,
+                "last_failure": now.isoformat() if failed else None,
+            }
```

Comment:
To keep the code DRY how about we write a function to create an entry so we can reuse it:
```python
def create_check_entry(name, message, failed):
    return {
        "name": name,
        "status": "CRITICAL" if failed else "OK",
        "status_code": 500 if failed else 200,
        "message": message,
        "last_checked": now.isoformat(),
        "last_success": now.isoformat() if not failed else None,
        "last_failure": now.isoformat() if failed else None,
    }
```

Reply:
oooh, I like it!